code/renderers/vue3/src/decorateStory.ts

```diff
@@ -64,8 +69,31 @@ export function decorateStory(
         return story;
       }
 
-      return prepare(decoratedStory, h(story, context.args)) as VueRenderer['storyResult'];
+      const innerStory = () => (story ? h(story, context.args) : null);
+      return prepare(decoratedStory, innerStory) as VueRenderer['storyResult'];
     },
     (context) => prepare(storyFn(context)) as LegacyStoryFn<VueRenderer>
   );
 }
+/**
+ * update the context with the update object from the decorator in reactive way
+ * @param context
+ * @param update
+ */
+export function updateReactiveContext(
+  context: StoryContext<VueRenderer>,
+  update: StoryContextUpdate<Partial<Args>> | undefined
+) {
+  if (update) {
+    const { args, argTypes } = update;
+    if (args && !argTypes) {
```

Comment:
What happens when `argTypes` is changed? (I don't know if this should be allowed anyway, by default we disallow `argTypes` updates: https://github.com/storybookjs/storybook/blob/2345f208aaa4477b9ce3f79e8942e4acfd7a00f2/code/lib/preview-api/src/modules/store/decorators.ts#L39

Reply:
Is it possible to compare the args to the previous values perhaps?